File: 03-06-2026/schema_validation1.js
function isValidSchema(obj) {
  return obj.hasOwnProperty("username") && typeof obj["username"] === "string";
}

console.log(isValidSchema({ username: "bob" })); // true.
console.log(isValidSchema({ username: "jen", posts: 30 })); // true.
console.log(isValidSchema({ username: "" })); // true.
console.log(isValidSchema({ username: 7 })); // false.
console.log(isValidSchema({ posts: 25 })); // false.

File: 03-06-2026/schema_validation1.py
def is_valid_schema(obj):
    if "username" not in obj or not isinstance(obj["username"], str):
        return False

    return True

print(is_valid_schema({ "username": "bob" })) # true.
print(is_valid_schema({ "username": "jen", "posts": 30 })) # true.
print(is_valid_schema({ "username": "" })) # true.
print(is_valid_schema({ "username": 7 })) # false.
print(is_valid_schema({ "posts": 25 })) # false.
